.github/workflows/ci-mac.yml

```diff
@@ -0,0 +1,126 @@
+# Copyright (C) 2022 Dremio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# Projectnessie GitHub Pull-Request / macOS CI
+
+name: macOS Build Check
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+  schedule:
+    # Run daily on week days
+    - cron:  '0 4 * * 1-5'
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  java:
+    name: Java/Gradle macOS
+    runs-on: macos-12
+    if: github.event.workflow_run.event != 'pull_request' || contains(github.event.pull_request.labels.*.name, 'pr-macos-win')
+    env:
+      SPARK_LOCAL_IP: localhost
+
+    steps:
+    - name: Setup Colima
+      timeout-minutes: 30
+      run: |
+        echo "::group::Install Podman"
+        brew install podman
+        echo "::endgroup::"
+
+        echo "::group::Initializing Podman"
+        podman machine init
+        echo "::endgroup::"
+
+        echo "::group::Starting Podman"
+        podman machine start
+        echo "::endgroup::"
+
+        echo "::group::Aliasing Podman"
+        ln -s /usr/local/bin/podman /usr/local/bin/docker
+        echo "::endgroup::"
+
+        export DOCKER_HOST="unix:///${HOME}/.local/share/containers/podman/machine/podman-machine-default/podman.sock"
+        echo "DOCKER_HOST=$DOCKER_HOST" >> ${GITHUB_ENV}
+
+        # See https://www.testcontainers.org/supported_docker_environment/
+        echo "TESTCONTAINERS_DOCKER_SOCKET_OVERRIDE=/var/run/docker.sock" >> ${GITHUB_ENV}
+
+        echo "::group::Docker info"
+        docker info
+        echo "::endgroup::"
+
+    - uses: actions/checkout@v3.3.0
+    - name: Setup Java, Gradle
+      uses: ./.github/actions/dev-tool-java
+
+    - name: Gradle / spotless
+      uses: gradle/gradle-build-action@v2
+      with:
+        # 'spotlessCheck' separate: workaround until
+        # https://github.com/diffplug/spotless/issues/1215 is fixed - should be merged w/ next step
+        arguments: spotlessCheck --scan
```

Comment:
Do we need a separate spotless check on Mac (and Win)? PR CI would validate the same source on Linux, would it not?

Reply:
Not really, removed it.